### ISSUE TITLE
Client Credentials not working in v1.2019.11.35

### ISSUE THREAD

**Issue by Splaxi** [MscrmTools/XrmToolBox#915]
I have been running on the v1.2019.7.34 without any issues.

After I updated to v1.2019.11.35 I'm am no longer able to logon to my D365CE instance. If I locate the old version from the backup folder and run it directly from there, it works flawlessly.

I have tried to recreate the connection string for my environment, but that didn't help.

![image](https://user-images.githubusercontent.com/8750181/72423066-e4609600-3783-11ea-9e65-b854b440fb48.png)

Looking at the connection log files for both versions, I can see the following being different:
For **v1.2019.7.34**
```
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Verbose	16	15-01-2020 10:26:53  Initialize CRM connection Started - AuthType: ExternalTokenManagement
```
For **v1.2019.11.35** 
```
Microsoft.Xrm.Tooling.Connector.CrmServiceClient	Verbose	16	15-01-2020 10:22:27  Initialize CRM connection Started - AuthType: ClientSecret
```

Notice that in the old version we are using and in the new version we are using 

Looking things up a bit on google, I quickly end up here:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.xrm.tooling.connector.authenticationtype?view=dynamics-xrmtooling-ce-9

I looked into the code of your connector "MscrmTools.Xrm.Connection" here on github and just replicated the connection from my local Visual Studio. I believe the first connection works, but something later in the code fails.

**Comment by peta** [comment]
I can confirm this error. After some live debugging of XrmToolbox I found out that the error is caused by an invalid token cache string value passed from `McTools.Xrm.Connection.ConnectionDetail.ConnectOAuth(...)` down to `Microsoft.Xrm.Tooling.Connector.CrmServiceClientTokenCache()`. 

It was introduced in [this commit](https://github.com/MscrmTools/MscrmTools.Xrm.Connection/commit/0892c2003fe74eeda5830e41b8d36fcddf2205f2#diff-2bc6f1b5fc9e268360486425b6453535)  and has already been fixed by a contributor end of november last year [in here](https://github.com/MscrmTools/MscrmTools.Xrm.Connection/commit/8204ce505cafced2e6d605abea695933e0db7683). But unfortunately even the current XrmToolbox prerelease references an [outdated version](https://github.com/MscrmTools/XrmToolBox/blob/6a3454a32625d35b3bc794780f8e6a1928f1aa64/XrmToolBox/packages.config#L13) of the `MscrmTools.Xrm.Connection` package which, unfortunately does not yet contain the fix from the commit mentioned above.

@MscrmTools  Any intents to update the deps and bring a new (pre-)release for this? I would consider it quite critical since S2S/App User scenarios raise in popularity.
